src/vsr/replica.zig

```diff
@@ -666,6 +719,12 @@ pub fn ReplicaType(
             });
             errdefer self.grid.deinit(allocator);
 
+            for (self.grid_write_blocks) |*block, i| {
+                errdefer for (self.grid_write_blocks[0..i]) |b| allocator.free(b);
+                block.* = try alloc_block(allocator);
+            }
+            errdefer for (self.grid_write_blocks) |b| allocator.free(b);
```

Comment:
I wonder if it makes sense to have

```zig
self.grid_write_blocks = alloc_blocks(allocator, constants.grid_repair_writes_max);
defere free_blocks(allocator, self.grid_write_blocks);
```

if only to avoid partial errdefers ?

I guess we can't easily express it in zig? We need "return array if size is comptime known, and a slice otherwise".

Reply:
We could make it `alloc_blocks(slice_of_blocks);` (`!void` return type), which would be compatible with arrays or slices (caller must heap-allocate the slice for non-comptime-known capacity). But I think this is a common enough Zig idiom that single allocation is ok, even if it is ugly.